language/vm/src/check_bounds.rs

```diff
@@ -388,112 +344,110 @@ impl<'a> BoundsChecker<'a> {
                 | GetTxnSenderAddress | GetTxnSequenceNumber | GetTxnPublicKey | Nop => (),
             }
         }
+        Ok(())
     }
 
-    fn check_type(&self, type_: &SignatureToken, errors: &mut Vec<VMStatus>) {
+    fn check_type(&self, type_: &SignatureToken) -> VMResult<()> {
         use self::SignatureToken::*;
 
         match type_ {
             Bool | U8 | U64 | U128 | Address | TypeParameter(_) => (),
-            Reference(ty) | MutableReference(ty) | Vector(ty) => self.check_type(ty, errors),
+            Reference(ty) | MutableReference(ty) | Vector(ty) => self.check_type(ty)?,
             Struct(idx) => {
-                check_bounds_impl(&self.module.struct_handles, *idx, errors);
+                check_bounds_impl(&self.module.struct_handles, *idx)?;
                 if let Some(sh) = self.module.struct_handles.get(idx.into_index()) {
                     if !sh.type_parameters.is_empty() {
-                        errors.push(
-                            VMStatus::new(StatusCode::NUMBER_OF_TYPE_ARGUMENTS_MISMATCH)
-                                .with_message(format!(
-                                    "expected {} type parameters got 0 (Struct)",
-                                    sh.type_parameters.len(),
-                                )),
-                        );
+                        return Err(VMStatus::new(StatusCode::NUMBER_OF_TYPE_ARGUMENTS_MISMATCH)
+                            .with_message(format!(
+                                "expected {} type parameters got 0 (Struct)",
+                                sh.type_parameters.len(),
+                            )));
                     }
                 }
             }
             StructInstantiation(idx, type_params) => {
-                check_bounds_impl(&self.module.struct_handles, *idx, errors);
+                check_bounds_impl(&self.module.struct_handles, *idx)?;
```

Comment:
@vgao1996 noticed that in my work for the binary format changes I missed checking the bounds on the `type_params` here. Do you want to add that? I think Victor may be doing a PR but just pointing out in case you wanted to do it and coordinate with him

Reply:
Added

Reply:
@vgao1996 notice above please